### PR TITLE
Update cypress, jest, eslint, and associated dependencies.

### PR DESCRIPTION
**Pull request by bgoldowsky**
Updates Cypress, Jest, Typescript, and ESLint, along with their related plugins and dependencies.

`tsconfig.json` and `.eslintrc.js` config files at the top level and in the `cypress` directory have been added or updated. The basis for these changes was primarily looking at the similar files in CLUE.

This should fix the problems we've been having with ESLint in VSCode flagging a lot of spurious errors.

It does, however, highlight some actual errors that we'll need to fix.

### CODE REVIEW

**Comment by bgoldowsky** [comment]
@tealefristoe could you check if these updates fix the ES Lint problems for you?
If it looks good then we can fix the remaining problems so that this PR can pass the tests.

**Comment by tealefristoe** [comment]
@bgoldowsky This fixes the lint errors I was seeing with svgs, video, and cypress. Great work!

**Comment by bgoldowsky** [comment]
I fixed the remaining lint errors.  Cypress test is failing but that should be fixed in a different PR.

File: cypress/e2e/workspace.test.ts [hunk 1]
```diff
@@ -36,10 +36,7 @@ context("Test the overall app", () => {
     {mode: Modes.simulation, organ: Organs.nose, title: "Immune Response Model Simulator"},
     {mode: Modes.simulation, organ: Organs.brain, title: "Focus Model Simulator"},
   ];
-  const modePages: PageInfo[] = [
-    {mode: Modes.animation, organ: Organs.heart, title: "Plaque Animation"},
-    {mode: Modes.simulation, organ: Organs.heart, title: "Plaque Model Simulator"}
-  ];
```
Comment (tealefristoe) [inline cypress/e2e/workspace.test.ts]: This was probably removed automatically, but it's being used again in one of my currently open PRs, so this change will cause a merge conflict.

**Review by tealefristoe — APPROVED** [review]
Looks good! I'm looking forward to having this all fixed up, especially for when working on cypress tests.

I'm a bit surprised to see that an an animation test is failing. What's going on there? What future PR will fix it?

**Pull request merged**